### ISSUE TITLE
SectionList scrolling is not possible when screen presentation: 'modal' with TransitionPresets.ModalPresentationIOS for Android devices

### ISSUE THREAD

**Issue by Marcin-Falkowski** [react-navigation/react-navigation#12257]
### Current behavior

Hi there. As I mentioned in title, I founded an issue with SectionList scrolling when the screen presentation: 'modal  with TransitionPresets.ModalPresentationIOS for Android devices. It looks really good on Android emulator but it's not working for real devices. I think the problem is related with "swap/scroll" gesture overlapping and I don't have an idea how to handle it.

Here is sample of code. If you need something more, please let me know in the comment.

**MainNavigation.tsx**
```
import {
  CardStyleInterpolators,
  createStackNavigator,
  TransitionPresets,
} from '@react-navigation/stack'
import {Modal} from 'react-native'

const Stack = createStackNavigator()

const MainNavigation = () => {
  return (
    <Stack.Navigator>
      <Stack.Screen
        name="Modal"
        component={Modal}
        options={{
          presentation: 'modal',
          headerShadowVisible: false,
          headerLeft: () => null,
          animationEnabled: true,
          cardStyleInterpolator: CardStyleInterpolators.forHorizontalIOS,
          headerTitleAlign: 'center',
          gestureEnabled: true,
          ...TransitionPresets.ModalPresentationIOS,
        }}
      />
    </Stack.Navigator>
  )
}
```

**Modal.tsx**

```
import {RefreshControl, SectionList, View} from 'react-native'

const Modal = () => {
  return (
    <View style={{flex: 1}}>
      <SectionList
        contentContainerStyle={{flexGrow: 1}}
        showsVerticalScrollIndicator={false}
        ItemSeparatorComponent={() => <View style={{height: 24}} />}
        onEndReachedThreshold={0.1}
        refreshControl={
          <RefreshControl refreshing={refreshing} onRefresh={onRefresh} />
        }
        onEndReached={onEndReached}
        onMomentumScrollBegin={onMomentumScrollBegin}
        sections={sections}
        keyExtractor={item => ((item && item.id + Math.random()) || 0) + ''}
        ListFooterComponent={ListFooterComponent}
        stickySectionHeadersEnabled={false}
        renderItem={renderItem}
        renderSectionHeader={renderSectionHeader}
      />
    </View>
  )
}
```

### Expected behavior

Expected behaviour is working as on IOS devices.

### Reproduction

https://snack.expo.dev/@uvision/sectionlist-bug

### Platform

- [x] Android
- [ ] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [x] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-top-tabs
- [x] @react-navigation/stack
- [x] @react-navigation/native-stack
- [ ] react-native-tab-view

### Environment

- [x] I've removed the packages that I don't use

values has been copied from package.json.
| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native         ^6.0.8      |
| @react-navigation/bottom-tabs     6.5.11     |
| @react-navigation/stack     ^6.2.0           |
| @react-navigation/native-stack   ^6.2.5      |
| react-native-safe-area-context   ^4.4.1      |
| react-native-screens       ^3.29.0            |
| react-native-gesture-handler      ^2.9.0     |
| react-native-reanimated. ^3.6.2|
| react-native            0.73.3               |
| node                      >=18             |
| npm or yarn               yarn             |

**Comment by Marcin-Falkowski** [comment]
> The versions mentioned in the issue for the following packages differ from the latest versions on npm:
> 
> * `@react-navigation/native` (found: `6.0.8`, latest: `7.0.2`)
> * `@react-navigation/bottom-tabs` (found: `6.5.11`, latest: `7.0.3`)
> * `@react-navigation/stack` (found: `6.2.0`, latest: `7.0.2`)
> 
> Can you verify that the issue still exists after upgrading to the latest versions of these packages?

Yes, issue still exists after upgrading to the latest version of these packages.

**Comment by gianlucalippolis** [comment]
I see two problems: 
- The first is that passing “gestureEnabled” might cause problems on scroll. 
- The second is that TransitionPresets.ModalPresentationIOS overrides cardStyleInterpolator.
So try removing “gestureEnabled”, “cardStyleInterpolator” and add TransitionPresets.ModalPresentationIOS only if the platform is iOS

**Comment by Marcin-Falkowski** [comment]
@gianlucalippolis Thank you for your response.

It's impossible. The clue of the problem is the iOS modal presentation style is required for Android as well. Settings as you mentioned are required to achieve that. There is no information I can't use ModalPresentationIOS for Android devices and it really works. gestureEnabled prop is required to "swipe-to-dismiss" action.

![Image](https://github.com/user-attachments/assets/4c519bf1-8c85-4ae3-b222-d0f96050acf8)
